src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/OutputProducer.cs

```diff
@@ -112,8 +113,12 @@ private Task FlushAsyncAwaited(WritableBufferAwaitable awaitable)
                     });
                 }
             }
+            await _flushTcs.Task;
 
-            return _flushTcs.Task;
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
I see, this is how you fixed the problem of "everyone having the same task". The CT is captured in the state machine and you check it on completion. Clever 👍